src/stirling/source_connectors/socket_tracer/conn_tracker.cc

```diff
@@ -797,12 +797,17 @@ void ConnTracker::IterationPreTick(
 
   // If remote_addr is missing, it means the connect/accept syscall was not traced.
   // Attempt to infer the connection information, to populate remote_addr and local_addr.
-  if (open_info_.remote_addr.family == SockAddrFamily::kUnspecified && socket_info_mgr != nullptr) {
-    InferConnInfo(proc_parser, socket_info_mgr);
+  const bool raddr_found = open_info_.remote_addr.family != SockAddrFamily::kUnspecified;
+  const bool laddr_found = open_info_.local_addr.family != SockAddrFamily::kUnspecified;
+  const bool info_mgr_ok = socket_info_mgr != nullptr;
 
+  if (!raddr_found && info_mgr_ok) {
+    InferConnInfo(proc_parser, socket_info_mgr);
     // TODO(oazizi): If connection resolves to SockAddr type "Other",
     //               we should mark the state in BPF to Other too, so BPF stops tracing.
     //               We should also mark the ConnTracker for death.
+  } else if (!laddr_found && info_mgr_ok) {
+    InferConnInfo(proc_parser, socket_info_mgr, true);
```

Comment:
I like the idea of checking for both missing local and remote addresses, though I am curious about something: 

Can you comment on how this is specific to long-lived connections where we miss the `connect`/`accept`? (as mentioned in the description).

If we've missed the initial connection, we should be missing both local and remote address. This change seems to be specific to the case where we've somehow caught the remote address, but not the local address from BPF. I don't doubt that it can happen, but I'm just trying to understand the scenario that triggered this.

Reply:
Thanks for the feedback @oazizi000. You're right that this is not specific to long-lived connections. I believe it is most likely to occur when we capture a `connect` syscall which lacks socket information for us trace the local IP from bpf. I've updated the description accordingly.